EDDI/EDDI.cs

```diff
@@ -607,128 +610,132 @@ private void keepAlive(string name, Action start)
             }
         }
 
-        public void eventHandler(Event journalEvent)
+        public void eventHandler(Event @event)
         {
-            if (journalEvent != null)
+            if (@event != null)
             {
                 try
                 {
-                    Logging.Debug("Handling event " + JsonConvert.SerializeObject(journalEvent));
+                    Logging.Debug("Handling event " + JsonConvert.SerializeObject(@event));
                     // We have some additional processing to do for a number of events
                     bool passEvent = true;
-                    if (journalEvent is FileHeaderEvent)
+                    if (@event is FileHeaderEvent)
+                    {
+                        passEvent = eventFileHeader((FileHeaderEvent)@event);
+                    }
+                    else if (@event is JumpedEvent)
                     {
-                        passEvent = eventFileHeader((FileHeaderEvent)journalEvent);
+                        passEvent = eventJumped((JumpedEvent)@event);
```

Comment:
These massive changes are a clue that we have the wrong architecture here. They are the equivalent of a massive switch statement where there should be a dispatch table or classical object-oriented virtual method

Reply:
I renamed `journalEvent` to `@event` in recognition of the fact that the status monitor and similar present opportunities for events that do not originate from the journal but otherwise I didn't substantially change the architecture. Can you give an example of what you have in mind?

Reply:
ah, now I look closer I see much of the code churn was due to the renaming of the governing variable. 

That illustrates a point in itself: code written according to the "SOLID" way of thinking doesn't exhibit much change upon a better variable renaming.

I take a few lessons here:

1. The big cascading if-else statement could have been isolated from outer variable names by being factored out into a separate method.

2. The big cascading if-else statement really wants to be a switch statement.

3. Generally, a wannabe big-arse switch statement is a subclass structure waiting to be unearthed.